qiskit/algorithms/gradients/base_estimator_gradient.py

```diff
@@ -166,6 +205,137 @@ def _validate_arguments(
                     f"not match the number of qubits of the {i}-th observable "
                     f"({observable.num_qubits})."
                 )
+        for i, (circuit, parameter) in enumerate(zip(circuits, parameters)):
+            if parameter is not None:
+                if not set(parameter).issubset(circuit.parameters):
+                    raise ValueError(
+                        f"The {i}-th parameter set contains parameters not present in the "
+                        f"{i}-th circuit."
+                    )
+
+    def _preprocess(self, circuits: Sequence[QuantumCircuit]):
```

Comment:
How about passing the `supported_gates` here as optional argument? That would allow us to remove it from the signature, which would be nice since it's not generally needed in the base.